src/transaction/buffer.rs

```diff
@@ -192,9 +209,24 @@ impl Buffer {
             .insert(key, BufferEntry::Put(value));
     }
 
+    /// Mark a value as Insert mutation into the buffer (does not write through).
+    pub async fn insert(&self, key: Key, value: Value) {
+        self.mutations
+            .lock()
+            .await
+            .insert(key, BufferEntry::Insert(value));
+    }
+
     /// Mark a value as deleted.
     pub async fn delete(&self, key: Key) {
-        self.mutations.lock().await.insert(key, BufferEntry::Del);
+        let mut mutations = self.mutations.lock().await;
+        let value = mutations.entry_map.entry(key).or_insert(BufferEntry::Del);
+
+        if let BufferEntry::Insert(_) = value {
+            *value = BufferEntry::CheckNotExist
+        } else {
+            *value = BufferEntry::Del
+        }
```

Comment:
`primary_key` is set in the `InnerBuffer::insert` method. We'd better use the `insert` method here

Reply:
Done